### ISSUE TITLE
Inputs inside list lose focus when arrow keys pressed

### ISSUE THREAD

**Issue by nwhittaker** [Esri/calcite-design-system#10271]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

Given a list of list items that contain input elements: When an input has focus and the user presses an `up` or `down` arrow key, focus can shift to the next list item instead of performing the input's default behavior.

### Expected Behavior

Focus is not shifted from the input and the input's default behavior is performed. In the case of text inputs, the cursor should move to the beginning or end of its value. For number inputs, the value should be incremented or decremented. Etc…

### Reproduction Sample

https://codepen.io/nwhittaker-esri/pen/OJeqrJq

### Reproduction Steps

1. Visit the repro.
2. Focus the first text input and hit either the `up` or `down` arrow key and see focus shifted away.
3. Focus the second input and hit the `down` arrow key and see the value is decremented but focus is also shifted the next list item. Interestingly, hitting the `up` arrow key correctly increments the value without shifting focus to the previous list item.

### Reproduction Version

2.12.2

### Priority impact

impact - p2 - want for an upcoming milestone

### Impact

A workaround is to ignore keyboard events on the list-item if the target is a descendant.

```ts
list.querySelector('calcite-list-item').addEventListener('keydown', event => {
  if (!event.target.matches('calcite-list-item')) {
    event.stopPropagation()
  }
})
```

However, this listener can't be delegated so it must be attached/detached to every list item. Which can become cumbersome for dynamically generated lists and easily lead to memory leaks.

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Field Apps

**Comment by driskull** [comment]
Input should be preventing default on ArrowDown just like it is on ArrowUp.

**Comment by driskull** [comment]
@geospatialem I don't think this is a bug. I think the author should be preventing key events in order to handle them.

**Comment by nwhittaker** [comment]
@driskull can you clarify the expected behavior? In the repro, the `up` and `down` keys don't behave the same. The `down` key behaves as observed and the `up` key behaves, in my opinion, as expected. Are you suggesting, the `down` key is instead behaving as expected and the `up` key is wrong?

Either way, one of the key behaviors is a bug, right?

**Comment by driskull** [comment]
@nwhittaker I see, you are correct. 

The `calcite-input-number` component calls `preventDefault()` when the arrow up key is pressed but not when the arrow down key is pressed. It should be calling `preventDefault()` with the down key as well.

@geospatialem this issue is valid, `calcite-input-number` should be fixed.

**Comment by geospatialem** [comment]
Verified with `3.2.0-next.12`

![Image](https://github.com/user-attachments/assets/bec1ffd6-26ac-45bb-81a8-b030ec1b2689)
